pkg/odo/cli/component/describe.go

```diff
@@ -35,15 +41,55 @@ var describeCmd = &cobra.Command{
 		componentDesc, err := component.GetComponentDesc(client, componentName, applicationName)
 		odoutil.LogErrorAndExit(err, "")
 
-		odoutil.PrintComponentInfo(componentName, componentDesc)
+		if outputFlag == "json" {
+			componentDef := getMachineReadableFormat(componentDesc)
+			out, err := json.Marshal(componentDef)
+			odoutil.LogErrorAndExit(err, "")
+			fmt.Println(string(out))
+		} else {
+
+			odoutil.PrintComponentInfo(componentName, componentDesc)
+		}
 	},
 }
 
+func getMachineReadableFormat(componentDesc component.Description) component.Component {
+	var urls []string
+	for _, url := range componentDesc.URLs {
+		urls = append(urls, url.Name)
+	}
+
+	var storage []string
+	for _, store := range componentDesc.Storage {
+		storage = append(storage, store.Name)
+	}
+
+	componentDef := component.Component{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Component",
+			APIVersion: "odo.openshift.io/v1beta1",
```

Comment:
I was wondering if its always guaranteed to be only this version
But if I am not wrong, we are using a specific client api even to fetch the list of components or do any operartion on components right?
But just wanted to be sure

Reply:
For now, it is OK to hardcode this. We are not really doing proper CRDs, this is just CRD like structure. 

@surajnarwade it might be better to use `v1alpha1` for now.

Reply:
yeah, `v1alpha1` sounds good